test/integration/formatting_stack/strategies.clj

```diff
@@ -3,26 +3,46 @@
    [clojure.java.shell :refer [sh]]
    [clojure.set :as set]
    [clojure.string :as string]
-   [clojure.test :refer [deftest is testing]]
+   [clojure.test :refer [deftest is testing use-fixtures]]
    [formatting-stack.strategies :as sut]
+   [formatting-stack.test-helpers :as test-helpers]
    [formatting-stack.util :refer [rcomp]]
    [nedap.speced.def :as speced])
   (:import
    (java.io File)))
 
-(def ^File deletable-file (File. "test-resources/deletable.clj"))
+(def ^File deletable-file (File. "git-integration-testing/test-resources/deletable.clj"))
 
-(def ^File rename-destination (File. "test-resources/deletable_moved.clj"))
+(def ^File rename-destination (File. "git-integration-testing/test-resources/deletable_moved.clj"))
 
-(def ^String createable-filename "test-resources/createable.clj")
+(def ^File creatable-filename (File. "git-integration-testing/test-resources/createable.clj"))
 
 (def creatable-contents (pr-str '(ns foo)))
 
-(assert (-> deletable-file .exists))
+(defn cleanup-testing-repo! []
+  (-> rename-destination .delete)
+  (-> creatable-filename .delete)
 
-(assert (not (-> rename-destination .exists)))
+  (sh "git" "reset" "--" "test-resources/deletable.clj")
+  (sh "git" "reset" "--" "test-resources/deletable_moved.clj")
+  (sh "git" "reset" "--" "test-resources/creatable.clj")
+  (sh "git" "checkout" "test-resources/deletable.clj")
 
-(assert (not (-> createable-filename File. .exists)))
+  (assert (-> deletable-file .exists))
+  (assert (not (-> rename-destination .exists)))
+  (assert (not (-> creatable-filename .exists))))
+
+(use-fixtures :once
+  test-helpers/with-git-repo
+
+  (fn [t]
+    (assert (-> deletable-file .exists))
+
+    (assert (not (-> rename-destination .exists)))
+
+    (assert (not (-> creatable-filename .exists)))
+
+    (t)))
 
 (defn assert-pristine-git-status! []
```

Comment:
Is `assert-pristine-git-status!` still useful?

(Might be part of the QA plan)

Reply:
While developing this PR did catch some nasty mistakes, so i'm happy to keep it in tbh.

Reply:
SGTM, I just wasn't sure (that is, without running it) whether it was still a runing a valid assertion